openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/schematypes/PrimitiveSchemaType.java

```diff
@@ -0,0 +1,99 @@
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import io.ballerina.compiler.syntax.tree.TypeDescriptorNode;
+import io.ballerina.openapi.exception.BallerinaOpenApiException;
+import io.ballerina.openapi.generators.schema.SchemaUtils;
+import io.swagger.v3.oas.models.media.Schema;
+
+import java.util.List;
+
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createIdentifierToken;
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createNodeList;
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createToken;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createMarkdownDocumentationNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createMetadataNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createSimpleNameReferenceNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createTypeDefinitionNode;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.PUBLIC_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.SEMICOLON_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.TYPE_KEYWORD;
+import static io.ballerina.openapi.generators.GeneratorConstants.NUMBER;
+import static io.ballerina.openapi.generators.GeneratorUtils.convertOpenAPITypeToBallerina;
+
+/**
+ * Generate TypeDefinitionNode and TypeDescriptorNode for primitive type schemas.
+ *
+ * @since 2.0.0
+ */
+public class PrimitiveSchemaType extends SchemaType {
+
+    private final boolean nullable;
+
+    public PrimitiveSchemaType(boolean nullable) {
+        this.nullable = nullable;
+    }
+
+    /**
+     * Generate TypeDefinitionNode for primitive types.
+     * -- ex:
+     * Sample OpenAPI :
+     * <pre>
+     *      components:
+     *          schemas:
+     *              PetName:
+     *                  type: string
+     *  </pre>
+     * Generated Ballerina type for the schema `PetName` :
+     * <pre>
+     *     public type PetName string;
+     * </pre>
+     */
+    @Override
+    public TypeDefinitionNode generateTypeDefinitionNode(Schema<Object> schemaValue, IdentifierToken typeName,
+                                                         List<Node> schemaDoc, List<AnnotationNode> typeAnnotations)
+            throws BallerinaOpenApiException {
+        MarkdownDocumentationNode documentationNode = createMarkdownDocumentationNode(
+                createNodeList(schemaDoc));
+        MetadataNode metadataNode = createMetadataNode(documentationNode, createNodeList(typeAnnotations));
+        return createTypeDefinitionNode(metadataNode, createToken(PUBLIC_KEYWORD), createToken(TYPE_KEYWORD),
+                typeName, this.generateTypeDescriptorNode(schemaValue), createToken(SEMICOLON_TOKEN));
+    }
+
+    /**
+     * Generate TypeDescriptorNode for primitive type schemas.
+     */
+    public TypeDescriptorNode generateTypeDescriptorNode(Schema schema) throws BallerinaOpenApiException {
```

Comment:
Don't we need to override this `generateTypeDescriptorNode`  there?

Reply:
Yes. Added